### PR TITLE
Move ceph container to quay.io/ceph/demo

### PR DESCRIPTION
**Pull request by fmount**
The ceph container has been recently migrated to a new image due to a refactoring of the repo [1].
This change align the Makefile to make sure we point to the right image [2].

[1] https://github.com/ceph/ceph-container/commit/2e0687a5147f931c5434614f3cd20f36fc6028da#diff-f6f28989a486d5db0bc66151752f59ad4a36d1b8d9f28b729609afa9f7c99014

[2] https://quay.io/repository/ceph/demo

Signed-off-by: Francesco Pantano <fpantano@redhat.com>

### CODE REVIEW

**Comment by fmount** [comment]
Moving to Draft as I might need to fix the script as well.

**Comment by fmount** [comment]
@stuggi can you double check it works properly for you?

**Comment by fultonj** [comment]
This change works for me. Thanks Francesco.

**Review by fultonj — COMMENTED** [review]
Works for me. I'll let Martin approve it though since he reported the initial issue.

**Review by stuggi — APPROVED** [review]
/lgtm  might be good to add in the Makefile  some help text for HOSTNETWORK, NETWORKS_ANNOTATION and MON_IP to explain them

File: Makefile
Comment (fmount) [inline Makefile:901]: `make ceph_help` will show a full helper

File: scripts/gen-ceph-kustomize.sh
Comment (fmount) [inline scripts/gen-ceph-kustomize.sh:192]: @stuggi we can either print a quick usage or a full help

**Comment by fmount** [comment]
> /lgtm might be good to add in the Makefile some help text for HOSTNETWORK, NETWORKS_ANNOTATION and MON_IP to explain them

Done

**Comment by fmount** [comment]
Also, we need this patch to unblock the CI testing [1], which is currently failing because of the ceph container [2].
@fultonj @stuggi are you ok to merge this patch?


[1] https://github.com/openshift/release/pull/36369
[2] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/36369/rehearse-36369-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy/1625530789926014976/artifacts/openstack-operator-build-deploy/openstack-k8s-operators-deploy/build-log.txt

**Pull request merged**
